src/internal/method.rs

```diff
@@ -54,6 +68,37 @@ where
     fn call<'a>(
         &'a self,
         arguments: <A as ArgumentsMatcher<'a>>::Arguments,
+    ) -> Result<R, CallError<'mock, 'a, A, R>> {
+        match self.call_order {
+            ExpectedCallOrder::Sequentially => {
+                self.handle_call_with_sequentially_ordered_expectations(arguments)
+            }
+            ExpectedCallOrder::Unordered => self.handle_call_with_unordered_expectations(arguments),
+        }
+    }
+
+    fn handle_call_with_sequentially_ordered_expectations<'a>(
```

Comment:
Can you test the new behaviour in this file also?

Reply:
Add at least a test that verifies that a method gives an error when no more calls are expected.

Reply:
87070c1